SimpleNotePad.java

```diff
@@ -0,0 +1,213 @@
+import javax.swing.text.StyledDocument;
+
+public class SimpleNotePad extends JFrame implements ActionListener{
+	JMenuBar menu = new JMenuBar();
+	JMenu fileMenu = new JMenu("File");
+	JMenu editMenu = new JMenu("Edit");
+	JTextPane textPane = new JTextPane();
+	JMenuItem newFile = new JMenuItem("New File");
+	JMenuItem saveFile = new JMenuItem("Save File");
+	JMenuItem printFile = new JMenuItem("Print File");
+	JMenuItem copy = new JMenuItem("Copy");
+	JMenuItem paste = new JMenuItem("Paste");
+
+	//JMenus/JMenuItems for new features
+	JMenuItem openFile = new JMenuItem("Open"); 
+	JMenu recentMenu = new JMenu("Recent"); 
+	JMenuItem simpleReplace = new JMenuItem("Replace");
+
+	LinkedList<File> recentFiles = new LinkedList<File>(); //holds recent files in order
+
+	public SimpleNotePad() {
+		setTitle("A Simple Notepad Tool");
+
+		fileMenu.add(openFile); //new feature
+		fileMenu.add(newFile);
+		fileMenu.add(saveFile);
+		fileMenu.add(printFile);
+		fileMenu.add(recentMenu); //new feature
+
+		editMenu.add(simpleReplace); //new feature 
+		editMenu.add(copy);
+		editMenu.add(paste);
+
+		//consolidated action listener and action command
+		newFile.addActionListener(this::action_newFile); 
+		saveFile.addActionListener(this::action_saveFile);       
+		printFile.addActionListener(this::action_printFile);        
+		copy.addActionListener(this::action_copy);        
+		paste.addActionListener(this::action_paste);        
+		openFile.addActionListener(this::action_openFile);        
+		simpleReplace.addActionListener(this::action_simpleReplace);
+
+		menu.add(fileMenu);
+		menu.add(editMenu);
+
+		setJMenuBar(menu);
+		add(new JScrollPane(textPane));
+		setPreferredSize(new Dimension(600,600));
+		setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+		setVisible(true);
+		pack();
+	}
+
+	public static void main(String[] args) {
+		SimpleNotePad app = new SimpleNotePad();
+	}
+
+	//extracted methods for each action command
+	@Override
+	public void actionPerformed(ActionEvent e) {
+
+	}
+
+	//new feature: simple-replace method
+	private void action_simpleReplace(ActionEvent e) {
+		String input = JOptionPane.showInputDialog(new JFrame("Input"), "Replace or insert with:");
+		if(input != null) textPane.replaceSelection(input);
+	}
+
+	//new feature: openRecent action method
+	private void action_openRecent(ActionEvent e) {
+		open(new File(e.getActionCommand())); 
+	}
+
+	//open method used by action_openRecent and action_openFile
+	private void open(File file) {
+		updateRecentFiles(file); //updates recent files
+		try {
+			BufferedReader input = new BufferedReader(new InputStreamReader(new FileInputStream(file)));
+			textPane.read(input, "Opening...");    
+		}    			
+		catch (IOException ex) {
+			JOptionPane.showMessageDialog(null,
+					"Open error" + ex, "Opening error",
+					JOptionPane.ERROR_MESSAGE);
+		}
+	}
+
+	//called by open() to update recent files list
+	private void updateRecentFiles(File file) {
```

Comment:
Good method to update your recentFiles LinkedList.  Nice use of if statements & loops.  To the point and easy to read